### DISCUSSION TITLE
edge attributes in heterogeneous graphs?

### DISCUSSION THREAD

**Discussion by paapu88** [pyg-team/pytorch_geometric#8685] [General]
Dear All,

I have been reading [this nice doc page](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html), and was wondering:

Are there any examples around where edge attributes would have been used when predicting/classifying with heterogeneous graphs?

Terveisin, Markus

**Comment by rusty1s** [comment]
We currently do not have such an example as best as I know, but edge features are well supported in heterogeneous graphs/heterogeneous GNNs in PyG. If you run into any trouble to integrate, I am happy to help.

**Reply by paapu88** [reply]
Hmm, for instance [here](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.HGTConv.html)
It seems to me that in forward there are no edge attributes.
Or have I misunderstood this?

**Reply by rusty1s** [reply]
Yes, `HGTConv` does not support edge features, but, e.g., `HeteroConv` or `to_hetero` does.

**Comment by paapu88** [comment]
Thanks!
I got confused by [this](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html) page at block:
```python
model = HeteroGNN(...)
output = model(data.x_dict, data.edge_index_dict, data.edge_attr_dict)
```
Because one gets the idea that there is some straightforward class 'HeteroGNN' which can digest 'data.edge_attr_dict'

Hope to have got it now. One might consider re-writing the documentation to make it more clear for beginners...
